### PR TITLE
Gracefully handle cases where IPv6 (AF_INET6) is not supported.

### PR DESCRIPTION
**Pull request by yossigo**
This fixes issue #1259 and allows a single configuration file to be used
both for IPv6 enabled and IPv6-less systems.

### CODE REVIEW

**Comment by cmouse** [comment]
Does leave the philosophical question open whether it should fail when it can't do what you ask or not.

**Comment by yossigo** [comment]
I think in this case the practical value of having a single configuration that "just works" justifies not doing what you ask, especially considering that no IPv6 support is probably more than an accidental typo. That's my 2 cents at least.

**Comment by Habbie** [comment]
What happens when the kernel has v6 but no interfaces have v6 addresses?

**Comment by yossigo** [comment]
Assuming the bound address is ::, server is up just like before this fix.

File: pdns/resolver.cc [hunk 1]
```diff
@@ -56,11 +56,14 @@ int makeQuerySocket(const ComboAddress& local, bool udpOrTCP)
   ComboAddress ourLocal(local);
   
   int sock=socket(ourLocal.sin4.sin_family, udpOrTCP ? SOCK_DGRAM : SOCK_STREAM, 0);
-  setCloseOnExec(sock);
   if(sock < 0) {
-    unixDie("Creating local resolver socket for "+ourLocal.toString() + ((local.sin4.sin_family == AF_INET6) ? ", does your OS miss IPv6?" : ""));
+    if(errno == EAFNOSUPPORT && local.sin4.sin_family == AF_INET6) {
+        return sock;
```
Comment (Habbie) [inline pdns/resolver.cc]: I think `return -1` might be clearer here.

**Comment by Habbie** [comment]
One nit, otherwise LGTM

**Pull request merged**